File: app/src/main/java/com/mudrahome/mhlms/Adapters/LeadListPagerAdapter.java
package com.mudrahome.mhlms.Adapters;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import com.mudrahome.mhlms.Fragments.LeadListFragment;
import com.mudrahome.mhlms.R;

public class LeadListPagerAdapter extends FragmentStatePagerAdapter {

    private int tabs;

    public LeadListPagerAdapter(FragmentManager fm, int tabs) {
        super(fm);
        this.tabs = tabs;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new LeadListFragment(R.string.admin);
            case 1:
                return new LeadListFragment(R.string.salesman);
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return tabs;
    }
}

File: app/src/main/java/com/mudrahome/mhlms/adapters/LeadListPagerAdapter.kt
package com.mudrahome.mhlms.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

//class LeadListPagerAdapter(fm: FragmentManager, private val userType: Int, private val tabs: Int) :
//    FragmentPagerAdapter(fm) {
//
//    override fun getItem(position: Int): Fragment? {
//        return if (userType == R.string.admin_and_salesman) {
//            when (position) {
//                0 -> LeadListFragment(R.string.admin)
//                1 -> LeadListFragment(R.string.salesman)
//                else -> null
//            }
//        } else {
//            when (position) {
//                0 -> LeadListFragment(R.string.telecaller)
//                1 -> LeadListFragment(R.string.teleassigner)
//                else -> null
//            }
//        }
//    }
//
//    override fun getCount(): Int {
//        return tabs
//    }
//}


class LeadListPagerAdapter(manager: FragmentManager) :
    FragmentPagerAdapter(manager) {

    private val fragmentList: MutableList<Fragment> = ArrayList()
    private val titleList: MutableList<String> = ArrayList()

    override fun getItem(position: Int): Fragment {
        return fragmentList[position]
    }

    override fun getCount(): Int {
        return fragmentList.size
    }

    fun addFragment(fragment: Fragment, title: String) {
        fragmentList.add(fragment)
        titleList.add(title)
    }

    override fun getPageTitle(position: Int): CharSequence? {
        return titleList[position]
    }
}
